### PR TITLE
Eliminates redundant `await`s.

### PR DESCRIPTION
**Pull request by scnale**
This also adds a couple of missing awaits in `handleRedeliveryWorkflow` and `handleDeliveryWorkflow`.

### CODE REVIEW

**Review by derpy-duck — APPROVED** [review]
Looks good to me

File: relayer_engine/src/plugin/src/plugin.ts [hunk 1]
```diff
@@ -560,7 +559,7 @@ export class GenericRelayerPlugin implements Plugin<WorkflowPayload> {
             return
           }
 
-          relayProvider
+          await relayProvider
```
Comment (JoeHowarth) [inline relayer_engine/src/plugin/src/plugin.ts:562]: Why should this await be removed? If we do so, then the following log line may fire when an error actually occurred here.

File: relayer_engine/src/plugin/src/plugin.ts [hunk 2]
```diff
@@ -604,7 +603,7 @@ export class GenericRelayerPlugin implements Plugin<WorkflowPayload> {
           relayerRefundAddress: wallet.address,
         }
 
-        relayProvider
+        await relayProvider
```
Comment (JoeHowarth) [inline relayer_engine/src/plugin/src/plugin.ts:606]: Same as above

**Review by JoeHowarth — CHANGES_REQUESTED** [review]

File: relayer_engine/src/plugin/src/plugin.ts (hunk 1 above)
Comment (scnale) [reply relayer_engine/src/plugin/src/plugin.ts:562]: Actually, this await is added :sweat_smile:. I wasn't sure if this was an intended dangling promise so I didn't want to merge this right away. Even if it is, I'd suggest adding a `.catch` to handle errors.

Comment (JoeHowarth) [reply relayer_engine/src/plugin/src/plugin.ts:562]: Good catch! And here I thought I was using too many awaits hahah

Yeah let's merge this

**Review by JoeHowarth — APPROVED** [review]

**Pull request merged**